src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -1092,6 +1092,14 @@ protected static ImmutableArray<string> CheckTupleElementNames(int cardinality,
             ImmutableArray<Location> memberLocations,
             ImmutableArray<bool> memberIsReadOnly);
 
+        /// <summary>
+        /// Returns true if there is an implicit (C#) or widening (VB) conversion from
+        /// <paramref name="fromType"/> to <paramref name="toType"/>. Returns false if
+        /// either <paramref name="fromType"/> or <paramref name="toType"/> is null, or
+        /// if no such conversion exists.
+        /// </summary>
+        public abstract bool HasImplicitConversion(ITypeSymbol fromType, ITypeSymbol toType);
```

Comment:
So, i'm on the fence about this entire PR :)   mainly because of how many *different* ways it seems to be able to determine this sort of thing.  i.e. we have:

CommonCompilation.HasImplicitConversion, then we have:
CSharpCompilation/BasicComplation.ClassifyConversion, then we have
IOperation.IConversionOperation/CommonConversion

My main issue is that it feels like this doesn't go far enough.  For example, what if i wanted ot ask "Is there a reference conversion between these two types?"  Now i'm back where i started. 

Please don't shoot me, but i think my preference here would be that instead of having this API we take the following two steps:

1. Move a ClassifyCommonConversion call down to Compilation (and have it return a CommonConversion).  
2. Add 'IsImplicit' to CommonConversion and document (like here) that it means 'implicit' for C# and 'widening' for VB.

This would seem to solve the motivating scenario, as someone could call ```compilation.ClassifyCommonConversion(...).IsImplicit```, but it would also still expose all the rest of hte information while being consistent with teh types/shapes we use elsewhere for talking about this stuff.

With this, we *could* still have ```HasImplicitConversion```.  But it could just be an extension method on Compilation, that called hte ClassifyCommonConversion helper.

Thoughts, @gafter?  This PR feels a bit too "one-off" to me.  I think we should go for a more generalized approach that fits in better with the shape of our API elsewhere.

Reply:
I think this is a great suggestion. I will investigate more.

Reply:
+1